docker/Dockerfile

```diff
@@ -1,71 +1,84 @@
 # syntax=docker.io/docker/dockerfile:1.4
 # layers for caching and versioning
 FROM cartesi/toolchain:0.14.0 as toolchain
-FROM cartesi/rootfs:0.17.0 as rootfs
 FROM cartesi/server-manager:0.7.0 as server-manager
-FROM cartesi/rollups-hardhat:0.9.0 as rollups-hardhat
+FROM cartesi/rollups-hardhat:0.9.1 as rollups-hardhat
 
 
-# download standard machine drives
+# stage to download dependencies
 FROM busybox as machine-core
 
 WORKDIR /opt/cartesi/share
 
-# download dependencies
 COPY dependencies .
 COPY shasumfile .
 RUN while read DEP; do wget -O $DEP; done < dependencies
 RUN sha1sum -c shasumfile
 
 
-# Deploy contracts locally
-FROM rollups-hardhat AS local-deployments-stage
+# stage to wrap the dapp installing resources needed to run inside the Cartesi Machine
+FROM dapp as wrapped-stage
 
-RUN sh -c "npx hardhat deploy --write true --network hardhat"
-ENTRYPOINT [ "/bin/sh" ]
+WORKDIR /
 
-# Download known testnet/mainet deployments
-FROM busybox as deployments-stage
+# normalize date for subsequent operations
+ENV SOURCE_DATE_EPOCH=1640995200
 
-WORKDIR /opt/cartesi/share
-COPY --from=machine-core opt/cartesi/share/rollups.tgz ./
-RUN tar -xzvf ./rollups.tgz
+# add the required 'dapp' user
+RUN useradd --create-home --user-group --uid 2874 dapp
```

Comment:
You will have permissions issues when trying to access `/dev/mtdblock1` as dapp user. The simple solution here is to not add the dapp usr, and let the dapp to be run as root. Running as dapp will require patching init script to change permissions of the device just after mounting for now. In a future machine SDK release this situation will improve.